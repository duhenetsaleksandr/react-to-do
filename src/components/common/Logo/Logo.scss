@use 'src/styles/colors';

.logo {
    flex: 1 0 auto;
    height: 30px;
    display: flex;
    align-items: center;
    svg {
        display: block;
        transition: all 0.2s ease;
    }

    &.light svg {
        color: colors.$secondary-color;
    }

    &.dark svg {
        color: colors.$primary-color;
    }
}
