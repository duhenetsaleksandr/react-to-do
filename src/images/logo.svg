<svg width="87" height="18" viewBox="0 0 87 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M2.85994 3.74667L9.93101 10.8177L7.10258 13.6462L0.0315124 6.5751L2.85994 3.74667Z" fill="#04D9B2"/>
<path d="M14.8258 0.219223L17.6542 3.04765L7.04763 13.6543L4.21921 10.8258L14.8258 0.219223Z" fill="#04D9B2"/>
<path d="M25.472 14H22V4.48003H25.528C26.9933 4.48003 28.0807 4.92336 28.79 5.81003C29.4993 6.6967 29.854 7.82136 29.854 9.18403C29.854 10.6587 29.4713 11.83 28.706 12.698C27.9407 13.566 26.8627 14 25.472 14ZM25.542 6.02003H23.806V12.474H25.444C26.2093 12.474 26.8207 12.222 27.278 11.718C27.7447 11.214 27.978 10.3694 27.978 9.18403C27.978 8.1947 27.782 7.42003 27.39 6.86003C26.998 6.30003 26.382 6.02003 25.542 6.02003Z" fill="#04D9B2"/>
<path d="M32.8937 11.97C32.8937 12.166 32.9731 12.3434 33.1318 12.502C33.2904 12.6514 33.5004 12.726 33.7617 12.726C34.0137 12.726 34.2657 12.67 34.5177 12.558C34.7791 12.446 35.0264 12.278 35.2598 12.054V11.018H33.8878C33.5704 11.018 33.3231 11.116 33.1458 11.312C32.9778 11.508 32.8937 11.7274 32.8937 11.97ZM33.7897 14.168C32.9591 14.168 32.3431 13.9674 31.9417 13.566C31.5404 13.1554 31.3398 12.6234 31.3398 11.97C31.3398 11.27 31.5777 10.7334 32.0537 10.36C32.5297 9.9867 33.1504 9.80003 33.9158 9.80003H35.2458C35.2364 9.4547 35.1337 9.1887 34.9377 9.00203C34.7511 8.80603 34.4897 8.70803 34.1537 8.70803C33.8364 8.70803 33.4911 8.75469 33.1178 8.84803C32.7444 8.93203 32.4318 9.0347 32.1798 9.15603L31.6898 7.64403C31.9511 7.4667 32.3477 7.32203 32.8797 7.21003C33.4211 7.0887 33.8364 7.02803 34.1257 7.02803C34.9377 7.02803 35.6331 7.26603 36.2118 7.74203C36.7904 8.2087 37.0798 8.80603 37.0798 9.53403V14H36.1418L35.6938 13.426C35.4418 13.622 35.1431 13.7947 34.7978 13.944C34.4618 14.0934 34.1257 14.168 33.7897 14.168Z" fill="#04D9B2"/>
<path d="M39.6645 4.20003C39.9538 4.20003 40.1965 4.29803 40.3925 4.49403C40.5978 4.69003 40.7005 4.93269 40.7005 5.22203C40.7005 5.51136 40.5978 5.75403 40.3925 5.95003C40.1965 6.14603 39.9538 6.24403 39.6645 6.24403C39.3845 6.24403 39.1465 6.14603 38.9505 5.95003C38.7545 5.75403 38.6565 5.51136 38.6565 5.22203C38.6565 4.93269 38.7545 4.69003 38.9505 4.49403C39.1465 4.29803 39.3845 4.20003 39.6645 4.20003ZM40.5745 7.21003V14H38.7545V7.21003H40.5745Z" fill="#04D9B2"/>
<path d="M44.2405 4.11603V14H42.4345V4.27003L44.2405 4.11603Z" fill="#04D9B2"/>
<path d="M47.3713 14.322L47.6933 13.524L45.2713 7.46203L47.1473 7.21003L48.7153 11.326L50.1993 7.21003H52.1173L49.0793 14.896C48.8646 15.4467 48.4866 15.918 47.9453 16.31C47.4039 16.702 46.8346 16.968 46.2373 17.108L45.5933 15.61C45.9199 15.47 46.2653 15.2974 46.6293 15.092C47.0026 14.896 47.2499 14.6394 47.3713 14.322Z" fill="#04D9B2"/>
<path d="M56.9333 8.79203H60.4053V13.496C60.0413 13.692 59.6026 13.86 59.0893 14C58.5759 14.1307 58.0253 14.196 57.4373 14.196C55.9533 14.196 54.8379 13.72 54.0913 12.768C53.3539 11.8067 52.9853 10.6447 52.9853 9.28203C52.9853 7.83536 53.4053 6.64069 54.2453 5.69803C55.0853 4.75536 56.2239 4.28403 57.6613 4.28403C57.9413 4.28403 58.338 4.3307 58.8513 4.42403C59.374 4.51736 59.8453 4.67603 60.2653 4.90003L59.6913 6.20203C59.2806 6.08069 58.9446 5.98736 58.6833 5.92203C58.4313 5.84736 58.0999 5.81003 57.6893 5.81003C56.8493 5.81003 56.1679 6.09936 55.6453 6.67803C55.1319 7.24736 54.8753 8.1107 54.8753 9.26803C54.8753 10.3134 55.0853 11.144 55.5053 11.76C55.9346 12.3667 56.5646 12.67 57.3953 12.67C57.7033 12.67 57.9693 12.656 58.1933 12.628C58.4173 12.5907 58.6133 12.5534 58.7813 12.516V10.192H56.7793L56.9333 8.79203Z" fill="#04D9B2"/>
<path d="M65.3788 14.168C64.3055 14.168 63.4655 13.846 62.8588 13.202C62.2615 12.5487 61.9628 11.676 61.9628 10.584C61.9628 9.50136 62.2755 8.63803 62.9008 7.99403C63.5355 7.35003 64.3988 7.02803 65.4908 7.02803C66.6201 7.02803 67.4928 7.36403 68.1088 8.03603C68.7341 8.69869 69.0468 9.57136 69.0468 10.654C69.0468 11.7367 68.7108 12.5954 68.0388 13.23C67.3668 13.8554 66.4801 14.168 65.3788 14.168ZM67.1708 10.654C67.1708 10.01 67.0261 9.50603 66.7368 9.14203C66.4568 8.7687 66.0555 8.58203 65.5328 8.58203C65.0008 8.58203 64.5854 8.74536 64.2868 9.07203C63.9881 9.3987 63.8388 9.89803 63.8388 10.57C63.8388 11.242 63.9648 11.7507 64.2168 12.096C64.4781 12.4414 64.8748 12.614 65.4068 12.614C65.9295 12.614 66.3541 12.46 66.6808 12.152C67.0074 11.8347 67.1708 11.3354 67.1708 10.654Z" fill="#04D9B2"/>
<path d="M71.8586 11.97C71.8586 12.166 71.9379 12.3434 72.0966 12.502C72.2553 12.6514 72.4653 12.726 72.7266 12.726C72.9786 12.726 73.2306 12.67 73.4826 12.558C73.7439 12.446 73.9913 12.278 74.2246 12.054V11.018H72.8526C72.5353 11.018 72.2879 11.116 72.1106 11.312C71.9426 11.508 71.8586 11.7274 71.8586 11.97ZM72.7546 14.168C71.9239 14.168 71.3079 13.9674 70.9066 13.566C70.5053 13.1554 70.3046 12.6234 70.3046 11.97C70.3046 11.27 70.5426 10.7334 71.0186 10.36C71.4946 9.9867 72.1153 9.80003 72.8806 9.80003H74.2106C74.2013 9.4547 74.0986 9.1887 73.9026 9.00203C73.7159 8.80603 73.4546 8.70803 73.1186 8.70803C72.8013 8.70803 72.4559 8.75469 72.0826 8.84803C71.7093 8.93203 71.3966 9.0347 71.1446 9.15603L70.6546 7.64403C70.9159 7.4667 71.3126 7.32203 71.8446 7.21003C72.3859 7.0887 72.8013 7.02803 73.0906 7.02803C73.9026 7.02803 74.5979 7.26603 75.1766 7.74203C75.7553 8.2087 76.0446 8.80603 76.0446 9.53403V14H75.1066L74.6586 13.426C74.4066 13.622 74.1079 13.7947 73.7626 13.944C73.4266 14.0934 73.0906 14.168 72.7546 14.168Z" fill="#04D9B2"/>
<path d="M79.6233 4.11603V14H77.8173V4.27003L79.6233 4.11603Z" fill="#04D9B2"/>
<path d="M85.4281 9.04403C85.1014 8.89469 84.7981 8.7827 84.5181 8.70803C84.2381 8.62403 83.9394 8.58203 83.6221 8.58203C83.3888 8.58203 83.2068 8.6287 83.0761 8.72203C82.9548 8.80603 82.8941 8.92736 82.8941 9.08603C82.8941 9.19803 82.9688 9.30536 83.1181 9.40803C83.2768 9.5107 83.5288 9.62269 83.8741 9.74403C84.4434 9.94936 84.9708 10.2387 85.4561 10.612C85.9414 10.976 86.1841 11.452 86.1841 12.04C86.1841 12.712 85.9414 13.2347 85.4561 13.608C84.9801 13.9814 84.2801 14.168 83.3561 14.168C83.0668 14.168 82.6794 14.112 82.1941 14C81.7181 13.888 81.3214 13.7107 81.0041 13.468L81.5501 12.152C81.8768 12.3014 82.1894 12.418 82.4881 12.502C82.7961 12.5767 83.0761 12.614 83.3281 12.614C83.6641 12.614 83.9208 12.5674 84.0981 12.474C84.2754 12.3807 84.3641 12.2594 84.3641 12.11C84.3641 11.9234 84.2754 11.7554 84.0981 11.606C83.9208 11.4567 83.6408 11.3214 83.2581 11.2C82.7168 11.0227 82.2034 10.7707 81.7181 10.444C81.2421 10.108 81.0041 9.66003 81.0041 9.10003C81.0041 8.45603 81.2421 7.95203 81.7181 7.58803C82.1941 7.21469 82.8381 7.02803 83.6501 7.02803C83.9394 7.02803 84.3128 7.0887 84.7701 7.21003C85.2274 7.33136 85.6288 7.50869 85.9741 7.74203L85.4281 9.04403Z" fill="#04D9B2"/>
</svg>
