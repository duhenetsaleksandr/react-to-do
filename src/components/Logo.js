import React from 'react';
import '../styles/logo.scss';
import logo from '../images/logo.svg';

export default function Logo() {
    return (
        <div className="logo">
            <img src={logo} alt="logo"/>
        </div>
    );
}
