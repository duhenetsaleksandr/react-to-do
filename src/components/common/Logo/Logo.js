import React, { useContext } from 'react';
import ThemeContext from "Context/ThemeContext";
import './Logo.scss';
import { ReactComponent as LogoSvg} from 'images/logo.svg';

const Logo = () => {
    const { darkTheme } = useContext(ThemeContext);
    return (
        <div className={`logo ${darkTheme ? 'light' : 'dark'}`}>
            <LogoSvg/>
        </div>
    );
};

export default Logo;
