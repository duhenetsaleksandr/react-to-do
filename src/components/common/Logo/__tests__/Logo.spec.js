import React from 'react';
import Logo from '../Logo';
import logo from 'images/logo.svg';

describe('should render Logo component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Logo/>);
    });

    it('should contain div.logo wrapper', () => {
        const wrapper = component.find('.logo');
        expect(wrapper).toHaveLength(1);
    });

    it('should contain img', () => {
        const image = component.find('svg');
        expect(image).toHaveLength(1);
    });

    it('should contain correct src img', () => {
        const image = component.find('svg');
        expect(image.getElement().props.children).toEqual(logo);
    });

    it('should render Logo component snapshot', () => {
        const component = shallow(<Logo/>);
        expect(component).toMatchSnapshot();
    });
});
