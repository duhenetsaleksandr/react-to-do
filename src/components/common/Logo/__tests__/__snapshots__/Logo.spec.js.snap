// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Logo component should render Logo component snapshot 1`] = `
<div
  className="logo light"
>
  <ForwardRef(SvgLogo) />
</div>
`;
